subsys/bluetooth/host/conn.c

```diff
@@ -562,6 +562,10 @@ static bool send_frag(struct bt_conn *conn, struct net_buf *buf, uint8_t flags,
 	k_sem_give(bt_conn_get_pkts(conn));
 	if (tx) {
 		conn_tx_destroy(conn, tx);
+		/* The buffer might not get destroyed, make sure that this tx context is not
+		 * re-used later.
+		 */
```

Comment:
Is this comment accurate? `conn_tx_destroy()` does not free the buffer, only the metadata. The buffer itself is freed by `net_buf_unref()`. As I understand it, we want to set `tx = NULL` so that `conn_tx_destroy()` is not called twice for the same `tx`.

Reply:
Then I think we agree about the reasoning here. Your updated comment is better.